Rewrite Community List component

ilmotta opened the pull request:
Fixes https://github.com/status-im/status-mobile/issues/16447

### Summary

This PR re-implements the component *Community List* according to guidelines and, most importantly, fixes a bunch of issues and tries to achieve 100% compatibility with Figma (it doesn't hurt to try right?).

The most important detail for you to know is that the component is trying to mirror Figma properties (also according to new guidelines written recently by @J-Son89). As a reviewer, you can basically open the Figma specs and follow each variant and play with the component in the Quo Preview screen.

Changelog:

- [x] Refactor to guidelines.
- [x] Fix: original implementation in `money/format-amounts` had a bug caught by new unit tests ("1000000" was formatted incorrectly).
- [x] Fix: Community permission tag correct background when theme is overridden and when blur is enabled.
- [x] Fix: Notification dot uses correct colors for dark mode and blur variants.
- [x] Fix: Community stats background when blur is enabled.
- [x] Fix: Community stats icon color when blur is enabled.
- [x] Add: The component's Quo preview screen is smarter and will only show form fields (aka descriptors) that are relevant for each component type (`engage`, `discover` or `share`). I think this significantly helps review the component.
- [x] Add: replace hardcoded community statistics with an implementation that actually accepts real numbers and which are correctly formatted.
- [x] Add: New Quo descriptor type `number`. Now we don't need to parse strings from the field type `text` :)
- [x] Add: Component supports all sorts of combinations of Figma properties, can't list them all here.
- [x] Add: Component uses correct shadows according to `foundations/shadows` (while doing this I found a bug I already have a fix https://github.com/status-im/status-mobile/issues/16526.
      
### Before vs after

In this PR I'm not sharing **before vs after** screenshots because the original implementation had too many limitations/bugs and this would just make things more confusing to you.

<img src="https://github.com/status-im/status-mobile/assets/46027/af3b7603-7df1-4293-a2a2-e692e7dab61f" width="500" />

### Notes

The PR will go over a designer's review after it has enough approvals. I'm curious to see how well they can test a component that more closely mirrors what's in Figma.

### Platforms

- Android
- iOS

### Areas that may be impacted

I believe there's no need for manual QA since the focus is on the Design System alone and all current usages of the component are not using the capabilities offered by the new implementation. Therefore, the e2e test suite, PR reviews and component specs should be enough.

### Steps to test

Open the Profile screen > `Quo 2.0 Preview` > `list-items` > `component-list`.

## Next steps after this PR is merged

- Have issues ready to be played for using the new component capabilities, for example in the Discover Communities screen.
- Move the old namespace `status-im.utils.money` to the new structure and remove the temporary duplication I'm creating in this PR.

status: ready

ilmotta committed:
Move community list component to proper Figma structure

ilmotta committed:
Rewrite component

ilmotta committed:
Add number field to quo preview

ilmotta committed:
Fix: permission tag blur variant

ilmotta committed:
Support subtitle prop

ilmotta committed:
Fix community stats icon colors with blur

ilmotta committed:
Fix permission tag blur color

ilmotta committed:
Fix blur colors

ilmotta committed:
Rename accessibility label

ilmotta committed:
Use different ns alias to avoid shadowing "theme"

ilmotta committed:
Remove unnecessary view component

ilmotta committed:
Show correct shadows

ilmotta committed:
Temporarily copy some money ns functions to new code

ilmotta committed:
Write component specs

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | 8f152460 | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16527/1/) | 2023-07-07 22:29:56 | ~2 min | `tests` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-16527/1/consoleText) |

ilmotta commented on src/utils/money.cljs:
```diff
@@ -0,0 +1,55 @@
+(ns utils.money
```
FYI: This namespace is temporarily being duplicated because I needed a function from the "old" code to format large amounts, but the linter fails if I require any namespace from `status_im` inside `quo2`.

I'm committed to work on a follow-up PR to move the entire `money` namespace to the new code structure since it's used in numerous places, it's worth it. As soon as this PR is merged ;)

ilmotta commented on src/status_im2/contexts/quo_preview/list_items/community_list.cljs:
```diff
@@ -0,0 +1,132 @@
+(ns status-im2.contexts.quo-preview.list-items.community-list
+  (:require [quo2.core :as quo]
+            [quo2.foundations.colors :as colors]
+            [quo2.theme :as quo.theme]
```
FYI: Aliasing as `quo.theme` to avoid having the namespace be shadowed by `theme` bindings.

ilmotta commented on src/react_native/core.cljs:
```diff
@@ -21,6 +21,7 @@
 (def text (reagent/adapt-react-class (.-Text ^js react-native)))
 (def text-input (reagent/adapt-react-class (.-TextInput ^js react-native)))
 
+(def pressable (reagent/adapt-react-class (.-Pressable ^js react-native)))
```
I think it's a good time for us to reconsider our usages of `TouchableHighlight`, `TouchableOpacity` and `TouchableWithoutFeedback` and use the new `Pressable` API.

Also, I've been thinking about talking to designers about micro-interactions for buttons and pressable "things". Might not even be hard depending on what they come up with and could bring a nice touch to the user experience.

ilmotta committed:
Fix lint

ibrkhalil commented on src/status_im2/contexts/quo_preview/preview.cljs:
```diff
@@ -199,6 +219,7 @@
        (case (:type desc)
          :boolean [customizer-boolean descriptor]
          :text    [customizer-text descriptor]
+         :number  [customizer-number descriptor]
          :select  [customizer-select descriptor]
```
Nitpick: Why not `type-customizer` for the name :)

ilmotta replied:
@ibrkhalil, if I could, I would rewrite the quo preview entirely, not because it's bad, but for lots of small things. But given that we're exploring Storybook now, I think it's probably not worth our time.

So why not `type-customizer`? Because I just wanted to follow the convention already in use here `customizer-XYZ`.

ibrkhalil reviewed (approved):
lgtm

J-Son89 commented on src/quo2/components/community/community_view.cljs:
```diff
@@ -1,16 +1,20 @@
 (ns quo2.components.community.community-view
-  (:require [quo2.components.community.style :as style]
+  (:require [quo.gesture-handler :as gesture-handler]
```
hmm, you should be using `react-native.gesture` instead? 🤔

ilmotta replied:
oh absolutely @J-Son89. This one slipped from the old code.

J-Son89 commented on src/quo2/components/community/community_view.cljs:
```diff
@@ -1,16 +1,20 @@
-  [rn/view (style/stats-count-container)
+  [{:keys [icon members-count icon-color accessibility-label]}]
+  [rn/view
+   {:accessibility-label accessibility-label
```
to make a view accessible you generally have to add `:accessibile true` to it as well. consider adding it.

ilmotta replied:
I agree with the technical importance of using `accessible`, but I don't think the mobile app cares about accessibility, like at all.

> When a view is marked as accessible, it is a good practice to set an accessibilityLabel on the view, so that people who use VoiceOver know what element they have selected.
>
> -- https://reactnative.dev/docs/accessibility#accessibilitylabel

We use accessibility labels for machines (e2e tests), not for humans. So in the end, if we really care about accessibility I think we (designers + devs) should go back to the drawing board and define some guidelines and ways of *testing accessibility* (e.g. using screen readers). It's a large topic for sure.

But as far as I remember @J-Son89, accessibility is not a priority until the product is launched. And in terms of color accessibility, I think our colors are not visually good for certain types of visual impairments (lack of contrast being an important one).

So in the end, why bother with `accessible` at this very moment if in the end the app is not accessible? 🤷🏼‍♂️

J-Son89 replied:
Actually for e2e tests it's needed. For jest tests it's not. But yeah you're right probably it's unnecessary to add unless it's required for e2e :)

J-Son89 commented on src/quo2/components/list_items/community_list/component_spec.cljs:
```diff
@@ -0,0 +1,79 @@
+            [test-helpers.component :as h]))
+
+(h/describe "Community list"
+  (h/test "default render"
```
I would say this test does more than the default render, i.e it tests both press and on-long press events are triggered as well.

ilmotta committed:
Use correct namespace from new code

ilmotta committed:
Add missing require to run component spec

ilmotta committed:
Remove some assertions from default rendering test

status-im-auto commented:
## 74% of end-end tests have passed
```
Total executed tests: 34
Failed tests: 9
Passed tests: 25
Not executed tests: 2
```
```
IDs of not executed tests: 703391,703297 
```
```
IDs of failed tests: 702784,702786,702731,702894,703086,702838,702783,702732,703133 
```
<h3>Not executed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=16527&TR_CASE_IDS=703391,703297">Rerun not executed tests</a></li></details><h3>Failed tests (9)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=develop&PR_ID=16527&TR_CASE_IDS=702784,702786,702731,702894,703086,702838,702783,702732,703133">Rerun failed tests</a></li><br/><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189514">test_restore_multiaccount_with_waku_backup_remove_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/test_public_chat_browsing.py:433: in test_restore_multiaccount_with_waku_backup_remove_switch
    self.errors.verify_no_errors()
base_test_case.py:182: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Contact(s) was (were) not restored from backup: MyCustomNickname!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a81cfe00252042f3ae9a47e9731a6a33?auth=d2e9ab10c28b7ba05b98f13b4c1ae4fe#117">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189491">test_1_1_chat_delete_via_long_press_relogin</a>, id: 702784 </b></td></tr><tr><td><p><blockquote><div>Device 2: Could not reach ChatsTab element by pressing back </div><div>Device 2: Find `ChatsTab` by `accessibility id`: `chats-stack-tab` </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1314: in test_1_1_chat_delete_via_long_press_relogin
    self.home_2.chats_tab.click()
../views/base_element.py:91: in click
    self.find_element().click()
../views/base_element.py:80: in find_element
    raise NoSuchElementException(
 Device 2: ChatsTab by accessibility id: `chats-stack-tab` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#389">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#342">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189484">test_1_1_chat_pin_messages</a>, id: 702731 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189490">test_1_1_chat_is_shown_message_sent_delivered_from_offline</a>, id: 702783 </b></td></tr><tr><td><p><blockquote><div># STEP: Device2 checks "Sending" status when sending message from offline</div><div>Device 2: Sending message 'test message' </div></blockquote></p>

```
critical/chats/test_1_1_public_chats.py:1287: in test_1_1_chat_is_shown_message_sent_delivered_from_offline
    self.chat_2.send_message(message_1)
../views/chat_view.py:963: in send_message
    self.chat_message_input.wait_for_element(wait_chat_input_sec)
../views/base_element.py:117: in wait_for_element
    raise TimeoutException(
 Device `2`: `ChatMessageInput` by` accessibility id`: `chat-message-input` is not found on the screen after wait_for_element
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#371">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#323">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189492">test_community_mentions_push_notification</a>, id: 702786 </b></td></tr><tr><td><p><blockquote><div># STEP: Admin gets push notification with the mention and tap it</div><div>Device 1: Getting PN by 'user_1' </div></blockquote></p>

```
critical/test_public_chat_browsing.py:949: in test_community_mentions_push_notification
    self.errors.verify_no_errors()
base_test_case.py:182: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 Push notification with the mention was not received by admin
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#405">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#310">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189508">test_community_contact_block_unblock_offline</a>, id: 702894 </b></td></tr><tr><td><p><blockquote><div>Device 1: Tap on found: AirplaneModeButton </div><div>Device 1: Find `Button` by `accessibility id`: `jump-to` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:789: in test_community_contact_block_unblock_offline
    self.home_1.jump_to_card_by_text('# %s' % self.channel_name)
../views/base_view.py:612: in jump_to_card_by_text
    self.jump_to_button.click()
../views/base_element.py:91: in click
    self.find_element().click()
../views/base_element.py:80: in find_element
    raise NoSuchElementException(
 Device 1: Button by accessibility id: `jump-to` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#337">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#290">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189513">test_community_mark_all_messages_as_read</a>, id: 703086 </b></td></tr><tr><td><p><blockquote><div>Device 1: Could not reach ChatsTab element by pressing back </div><div>Device 1: Find `CommunitiesTab` by `accessibility id`: `communities-stack-tab` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:848: in test_community_mark_all_messages_as_read
    self.home_1.communities_tab.click()
../views/base_element.py:91: in click
    self.find_element().click()
../views/base_element.py:80: in find_element
    raise NoSuchElementException(
 Device 1: CommunitiesTab by accessibility id: `communities-stack-tab` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#388">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#307">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189496">test_community_message_send_check_timestamps_sender_username</a>, id: 702838 </b></td></tr><tr><td><p><blockquote><div>Device 2: Looking for a message by text: new message </div><div>Device 2: Find `Username` by `xpath`: `//*[starts-with(@text,'new message')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']/android.view.ViewGroup/android.widget.TextView[1]` </div></blockquote></p>

```
critical/test_public_chat_browsing.py:493: in test_community_message_send_check_timestamps_sender_username
    if self.channel_2.chat_element_by_text(new_message).username.text != self.username_1:
../views/base_element.py:391: in text
    text = self.find_element().text
../views/base_element.py:80: in find_element
    raise NoSuchElementException(
 Device 2: Username by xpath: `//*[starts-with(@text,'new message')]/ancestor::android.view.ViewGroup[@content-desc='chat-item']/android.view.ViewGroup/android.widget.TextView[1]` is not found on the screen
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#144">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#103">Steps, video, logs</a></li><li><a href="https://media.giphy.com/media/9M5jK4GXmD5o1irGrF/giphy.gif">Failure screenshot</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189485">test_group_chat_pin_messages</a>, id: 702732 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Pin feature is in development]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (25)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189507">test_community_one_image_send_reply</a>, id: 702859 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#205">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#155">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189500">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#162">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#114">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189501">test_community_links_with_previews_github_youtube_twitter_gif_send_enable</a>, id: 702844 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#273">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#245">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189515">test_community_several_images_send_reply</a>, id: 703194 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#231">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#193">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189502">test_community_leave</a>, id: 702845 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#434">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#333">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189499">test_community_unread_messages_badge</a>, id: 702841 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#312">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#282">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>7. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189498">test_community_emoji_send_copy_paste_reply</a>, id: 702840 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#252">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#230">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>8. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189497">test_community_message_delete</a>, id: 702839 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/9ea38a7fa3414ca0877148f2733f3678?auth=9d339e6b163e14a08cae7a0431123100#186">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/10b7da07d1c74e98b5036deae4a3eec9?auth=24582bba68584b1e9849737a6120e717#126">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestGroupChatMultipleDeviceMergedNewUI:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189493">test_group_chat_join_send_text_messages_push</a>, id: 702807 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/17713bad7b5f4957a68038e413a723bc?auth=935b99f1662e0fc47cb2007ee0de4dc7#122">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7b350b04d3084de98a06e3f0a47302ba?auth=c90f47b384d12950a8bc6bd0f9724001#68">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a99a48c1bad4e178e81ff99340db961?auth=8049d131c928a99d66ae6a522a4186ae#68">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189494">test_group_chat_offline_pn</a>, id: 702808 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/17713bad7b5f4957a68038e413a723bc?auth=935b99f1662e0fc47cb2007ee0de4dc7#237">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7b350b04d3084de98a06e3f0a47302ba?auth=c90f47b384d12950a8bc6bd0f9724001#147">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a99a48c1bad4e178e81ff99340db961?auth=8049d131c928a99d66ae6a522a4186ae#116">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189516">test_group_chat_reactions</a>, id: 703202 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/17713bad7b5f4957a68038e413a723bc?auth=935b99f1662e0fc47cb2007ee0de4dc7#136">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/7b350b04d3084de98a06e3f0a47302ba?auth=c90f47b384d12950a8bc6bd0f9724001#93">Steps, video, logs</a></li></ul></p><p>Device 3:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/8a99a48c1bad4e178e81ff99340db961?auth=8049d131c928a99d66ae6a522a4186ae#73">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189503">test_community_navigate_to_channel_when_relaunch</a>, id: 702846 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a81cfe00252042f3ae9a47e9731a6a33?auth=d2e9ab10c28b7ba05b98f13b4c1ae4fe#74">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189487">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a81cfe00252042f3ae9a47e9731a6a33?auth=d2e9ab10c28b7ba05b98f13b4c1ae4fe#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189510">test_activity_center_reply_read_unread_delete_filter_swipe</a>, id: 702947 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f71deadbaec4e69a14874574fc2ef41?auth=b584c9bfa6575d8e929081c250086976#163">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d820101d0a554aabba9ad60621729d37?auth=44fe29760513917e8fbe75c33bf8ec7e#106">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189509">test_navigation_jump_to</a>, id: 702936 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f71deadbaec4e69a14874574fc2ef41?auth=b584c9bfa6575d8e929081c250086976#142">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d820101d0a554aabba9ad60621729d37?auth=44fe29760513917e8fbe75c33bf8ec7e#103">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189511">test_activity_center_mentions</a>, id: 702957 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f71deadbaec4e69a14874574fc2ef41?auth=b584c9bfa6575d8e929081c250086976#275">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d820101d0a554aabba9ad60621729d37?auth=44fe29760513917e8fbe75c33bf8ec7e#122">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189512">test_activity_center_admin_notification_accept_swipe</a>, id: 702958 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/3f71deadbaec4e69a14874574fc2ef41?auth=b584c9bfa6575d8e929081c250086976#311">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/d820101d0a554aabba9ad60621729d37?auth=44fe29760513917e8fbe75c33bf8ec7e#140">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189486">test_1_1_chat_text_message_delete_push_disappear</a>, id: 702733 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#320">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#257">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189483">test_1_1_chat_message_reaction</a>, id: 702730 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#76">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#73">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>3. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189495">test_1_1_chat_push_emoji</a>, id: 702813 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#251">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#189">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>4. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189488">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#190">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#145">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>5. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189506">test_1_1_chat_edit_message</a>, id: 702855 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#294">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#216">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>6. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189489">test_1_1_chat_emoji_send_reply_and_open_link</a>, id: 702782 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a0862b5780b24781bf86b2551faeab72?auth=d7fe06d7acbd4eec2d426fe1d94e7382#128">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/5604a80e078c4c9e9180ac295684e3f2?auth=6819563094eba5b31e28a95161d124d5#104">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestActivityCenterContactRequestMultipleDevicePR:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189504">test_activity_center_contact_request_decline</a>, id: 702850 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a6d55e6228f2459584bfd275e7cc350e?auth=e85440f8d358c69b8f7e7f5a5725d55b#55">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cc62633401214dcaa5e225cb1b714393?auth=9cf23091008d2f6d7c6dae88c5c630e0#55">Steps, video, logs</a></li></ul></p></ul></p></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3189505">test_activity_center_contact_request_accept_swipe_mark_all_as_read</a>, id: 702851 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/a6d55e6228f2459584bfd275e7cc350e?auth=e85440f8d358c69b8f7e7f5a5725d55b#88">Steps, video, logs</a></li></ul></p><p>Device 2:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/cc62633401214dcaa5e225cb1b714393?auth=9cf23091008d2f6d7c6dae88c5c630e0#92">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

churik commented:
@Francesca-G can you please review this PR from design perspective?
Thanks!

ilmotta commented:
> @Francesca-G can you please review this PR from design perspective? Thanks!

Hey @churik, thanks for asking for her review. Just one thing, before asking for the review I was going to explain a little bit how she can review this type of component, because this component in particular doesn't have enough usages/variations in real screens, but it does have a ton of variations in the Design System. So unless the component is reviewed using the Quo Preview screen (which I built to have all variations) the review would be sort of pointless.

@Francesca-G, as I mentioned to you yesterday off-GitHub, non-production builds, e.g. PR builds have some features enabled that are never available to end-users. One such feature is named `Quo2.0 Preview`. `Quo` because the name of the Design System library we're coding is called `quo` and `2.0` because it's the revamped mobile app. You can access this feature from the profile screen.

<img src="https://github.com/status-im/status-mobile/assets/46027/aa2512c6-ada3-41de-821f-65796ae7ea65" height="400" />

In the next screen you'll see the list of components. Most of the time, the names of the components and the categories (groups) use the same names used by the Design System.

<img src="https://github.com/status-im/status-mobile/assets/46027/e7b96e49-4baf-4420-a0de-f46790ffe43e" height="600" />

<br>

For this PR, I rewrote the Community List component ([Figma specs](https://www.figma.com/file/WQZcp6S0EnzxdTL4taoKDv/Design-System-for-Mobile?type=design&node-id=8917-153680&mode=design&t=GHNFCReTgStqAEot-0)) and tried to mirror as much as possible all Figma properties and their variations in a way that you or a QA/developer can much more easily walk through each variation in Figma and do the same in the mobile PR build.

The component is indeed used in a real screen, the Discover Communities screen, but it uses only the most basic variation of the component, so I highly recommend you focus your review on the Preview screen because that's where you'll find the bugs :)

There's an ongoing effort to start using Storybook for mobile, which will, in practice, replace our in-house preview screens and will simplify the whole process of reviewing DS components, but it's still alpha stage, so it'll take some time to iron out some things and migrate to Storybook.

That's all :) Thank you

Francesca-G commented:
@ilmotta Thank you for clarifying how to use Quo Previews!
I'm preparing a more in-depth review about various components :)

@churik in order to review the community list component I think I will need more options: as you can see from the screen below the notification options for mention counts and unread messages don't really change the preview (or maybe it's just me, the lock and the status of the lock both work just fine). 
I'd also like to have an option to display more list items in order to check the spacing between the elements, since this is a list component. Thanks!

![IMG_0084](https://github.com/status-im/status-mobile/assets/122543089/b9226620-3622-4ba6-97f4-0bc8c8a8e293)

churik commented:
@ilmotta thank you for clarifying and sorry if I was too fast with the design review!  
Next time I'll leave your PRs for you to decide.

J-Son89 commented:
> @Francesca-G, I'll try to expand on your comments.
> 
> 
> 
> > I'd also like to have an option to display more list items in order to check the spacing between the elements, since this is a list component.
> 
> 
> 
> The component in the DS is a single item replicated multiple times and grouped by the type of the variation (e.g. `Discover`). Therefore, in the code, we need to mirror that, and that's why you can only customize a single item in the Preview.
> 
> 
> 
> Still on this topic, I think the DS component should be named just `Community`, not `Community List`, just like in the same Figma page `Channel` is used instead of `Channel List` because none of the components in that screen are lists (they are list items). If we do that in Figma, I'll also rename in the code (it's a 1min thing).
> 
> 
> 
> > in order to review the community list component I think I will need more options: as you can see from the screen below the notification options for mention counts and unread messages don't really change the preview (or maybe it's just me, the lock and the status of the lock both work just fine).
> 
> 
> 
> I think you're testing a different component. The one this PR has implemented is accessed in `Preview` > `list-items` > `community-list` and it has many different options.
> 
> 
> 
> <img src="https://github.com/status-im/status-mobile/assets/46027/b8511c04-33fd-4805-8cef-3f3e4f5d755b" height="400" />
> 
> 

+1 there are multiple components in the design system misnamed, ie it's a list item and it's called *-list and the opposite too, for example category in DS is a list component. It just confuses devs 🙃

Francesca-G reviewed (commented):
@ilmotta thank you so much! 
I renamed the DS component to just **Community** instead of **Community List** and apologies, I was testing a different component as you mentioned. 

Here's the [Figma review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3374%3A300992&mode=design&t=j4895iAGv2jfi45D-1) for the Community List component :)

J-Son89 commented:
> @ilmotta thank you so much! 
> 
> I renamed the DS component to just **Community** instead of **Community List** and apologies, I was testing a different component as you mentioned. 
> 
> 
> 
> Here's the [Figma review](https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=3374%3A300992&mode=design&t=j4895iAGv2jfi45D-1) for the Community List component :)

@ilmotta @Francesca-G why not community-list-item :)

ilmotta commented:
> @ilmotta thank you for clarifying and sorry if I was too fast with the design review! Next time I'll leave your PRs for you to decide.

I totally understand @churik, no worries.

<img src="https://github.com/status-im/status-mobile/assets/46027/8e758812-b1a2-4522-99ab-37b3448857be" height="300" />

> I renamed the DS component to just Community instead of Community List and apologies, I was testing a different component as you mentioned.

Thank you @Francesca-G. Swift review!

> @ilmotta @Francesca-G why not community-list-item :)

@J-Son89, I think in the code that's how it should be named 👍🏼 and that's how I'm going to rename it now, but I guess in Figma it could be just `Community` since it's already inside the page `List items` (?) Wdyt?

J-Son89 commented:
Oh no then I think it's fine 👍

ilmotta committed:
Rename component to mirror what's in Figma

ilmotta committed:
Fix remaining community stats UI bugs

ilmotta commented:
@Francesca-G, I have fixed 3 of the 4 bugs you reported. The one I didn't fix is because it's a more complicated fix in the permission tag component, so I created a separate issue https://github.com/status-im/status-mobile/issues/16566.

But I fixed all the 3 UI bugs you reported about the community statistics component.

- [x] Fixed: Label text size looks bigger than design https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3374:300992&mode=design#494883189.
- [x] Fixed: Outdated active members icon https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3374:300992&mode=design#494884965.
- [x] Fixed: Too much space between active members icon and label https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3374:300992&mode=design#494883483

So all that's left is your final review and PR approval if there are no outstanding issues 🌞 ✈️ Thank you

Francesca-G reviewed (approved):
@ilmotta everything you fixed seems good now :) 
Thank you for creating the separate issue to update the token tag design.
